hera_sim/visibilities/simulators.py

```diff
@@ -13,374 +12,129 @@
-        ):
-            raise ValueError(
-                "sky_intensity has a different number of freqs " "than sky_freqs."
-            )
-
-        if self.sky_intensity is not None and self.sky_intensity.ndim != 2:
-            raise ValueError(
-                "sky_intensity must be a 2D array (a healpix map " "per frequency)."
-            )
-
-        if not (self.point_source_ability or self.point_source_pos is None):
-            warnings.warn(
-                "This visibility simulator is unable to explicitly "
-                "simulate point sources. Adding point sources to "
-                "diffuse pixels."
-            )
-            if self.sky_intensity is None:
-                self.sky_intensity = 0
-            self.sky_intensity += self.convert_point_sources_to_healpix(
-                self.point_source_pos, self.point_source_flux, self.nside
-            )
-
-        if not self.diffuse_ability and self.sky_intensity is not None:
-            warnings.warn(
-                "This visibility simulator is unable to explicitly "
-                "simulate diffuse structure. Converting diffuse "
-                "intensity to approximate points."
-            )
-
-            (pos, flux) = self.convert_healpix_to_point_sources(self.sky_intensity)
-
-            if self.point_source_pos is None:
-                self.point_source_pos = pos
-                self.point_source_flux = flux
-            else:
-                self.point_source_flux = np.hstack((self.point_source_flux, flux))
-                self.point_source_pos = np.hstack((self.point_source_pos, pos))
-
-            self.sky_intensity = None
-
-    @staticmethod
-    def convert_point_sources_to_healpix(
-        point_source_pos, point_source_flux, nside=2 ** 5
-    ):
-        """
-        Convert point sources to an approximate diffuse HEALPix model.
-
-        The healpix map returned is in RING scheme.
-
-        Parameters
-        ----------
-        point_source_pos : array_like
-            An array of point sources. For each source, the entries are
-            (ra, dec) [rad] (assumed to be in J2000).
-            Shape=(N_SOURCES, 2).
-        point_source_flux : array_like
-            point_source_flux : array_like, optional
-            An array of fluxes of the given point sources, per
-            frequency. Fluxes in [Jy]. Shape=(NFREQS, N_SOURCES).
-        nside : int, optional
-            HEALPix nside parameter (must be a power of 2).
-
-        Returns
-        -------
-        array_like
-            The HEALPix diffuse model. Shape=(NFREQ, NPIX).
-        """
-        hmap = np.zeros((len(point_source_flux), aph.nside_to_npix(nside)))
-
-        # Get which pixel every point source lies in.
-        pix = aph.lonlat_to_healpix(
-            lon=point_source_pos[:, 0] * units.rad,
-            lat=point_source_pos[:, 1] * units.rad,
-            nside=nside,
+        else:
+            self.beam_ids = np.array(beam_ids, dtype=int)
```

Comment:
I was actually considering whether we instead preferred to keep the beam_ids dict from pyuvsim instead of the list that we turn it into. Or perhaps keep both around on the instance. We can allow both as input at least, good point.

The `ModelData` class is going to be used by the user to build up a `VisibilitySimulation`.

Reply:
I've updated accept the dict input. Will await your ideas on whether we should keep the dict format around (or switch to it).

Reply:
Another thing to think about: in the the `from_config` method, we do this to get the `beam_ids` from the dict:

```python
nms = list(uvdata.antenna_names)
tmp_ids = np.zeros(len(beam_ids), dtype=int)
for name, beam_id in beam_ids.items():
    tmp_ids[nms.index(name)] = beam_id
beam_ids=tmp_ids
```

Instead of simply `np.array(list(beam_ids.values()), dtype=int)`. I wonder if we did this for a reason -- perhaps we were ensuring ordering for pre-py36, where dicts weren't ensured to have a particular ordering?

Reply:
Just to be sure, I changed the conversion in `_process_beams` to take account of this ordering, but simplified the above into this:

```python
beam_ids = np.array(
    [beam_ids[nm] for nm in self.uvdata.antenna_names], dtype=int
)
```

Reply:
Nice, this is much more compact. I'm not sure why it was done the way it was before, but this looks like it achieves the same thing.

As for whether we should use the dict format from `pyuvsim` or not... I think I need to have a better picture of how everything works before weighing in on that.